lib/srv/db/access_test.go

```diff
@@ -2637,6 +2639,19 @@ func (c *testContext) setupDatabaseServer(ctx context.Context, t testing.TB, p a
 				t.Fatal("timed out waiting for inventory handle sender")
 			}
 		}
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
+			dbServers, err := c.proxyServer.cfg.AccessPoint.GetDatabaseServers(ctx, apidefaults.Namespace)
+			if !assert.NoError(t, err) {
+				return
+			}
+			names := map[string]struct{}{}
+			for _, srv := range dbServers {
+				names[srv.GetName()] = struct{}{}
+			}
+			for _, db := range p.Databases {
+				assert.Contains(t, names, db.GetName())
+			}
```

Comment:
nit: probably can do something like this
```golang
			assert.ElementsMatch(t,
				slices.Collect(types.ResourceNames(p.Databases)),
				slices.Collect(types.ResourceNames(dbServers)),
			)
```
doesn't save a lot of lines but slightly easier to read